<?xml version="1.0"?>
<launch>
    <arg name="launch_hw" default="true"/>
    <arg name="sensor_model" default="ARS548"/>
    <arg name="frame_id"  default="continental"/>
    <arg name="base_frame"  default="base_link"/>
    <arg name="object_frame"  default="base_link"/>

    <arg name="odometry_topic" default="odometry_input"/>
    <arg name="acceleration_topic"  default="acceleration_input"/>
    <arg name="steering_angle_topic"  default="steering_angle_input"/>

    <!-- Ethernet based sensor: ARS548 -->
    <arg name="sensor_ip" default="10.13.1.114" description="Radar Sensor IP"/>
    <arg name="host_ip" default="10.13.1.166" description="Broadcast IP from Sensor"/>
    <arg name="multicast_ip" default="224.0.2.2" description="Multicast Sensor IP"/>
    <arg name="data_port" default="42102" description="Radar Data Port"/>
    <arg name="configuration_host_port" default="42401" description="Radar host configuration port"/>
    <arg name="configuration_sensor_port" default="42101" description="Radar sensor configuration port"/>
    <arg name="use_sensor_time" default="false" description="Whether to use or not the timestamp from the sensor"/>

    <arg name="configuration_vehicle_length" default="4.89" description="New vehicle length"/>
    <arg name="configuration_vehicle_width" default="1.896" description="New vehicle width"/>
    <arg name="configuration_vehicle_height" default="2.5" description="New vehicle height"/>
    <arg name="configuration_vehicle_wheelbase" default="2.79" description="New vehicle wheel base"/>

    <group if="$(eval &quot;'$(var sensor_model)' == 'ARS548' &quot;)" >

        <node pkg="nebula_ros" exec="continental_ars548_ros_wrapper_node" name="nebula_continental_ars548" output="screen">

            <remap from="/diagnostics" to="diagnostics"/>
            <remap from="odometry_input" to="$(var odometry_topic)"/>
            <remap from="acceleration_input" to="$(var acceleration_topic)"/>
            <remap from="steering_angle_input" to="$(var steering_angle_topic)"/>

            <param name="launch_hw" value="$(var launch_hw)"/>
            <param name="sensor_model" value="$(var sensor_model)"/>
            <param name="frame_id" value="$(var frame_id)"/>
            <param name="base_frame" value="$(var base_frame)"/>
            <param name="object_frame" value="$(var object_frame)"/>
            <param name="host_ip" value="$(var host_ip)"/>
            <param name="multicast_ip" value="$(var multicast_ip)"/>
            <param name="data_port" value="$(var data_port)"/>
            <param name="sensor_ip" value="$(var sensor_ip)"/>
            <param name="configuration_host_port" value="$(var configuration_host_port)"/>
            <param name="configuration_sensor_port" value="$(var configuration_sensor_port)"/>
            <param name="use_sensor_time" value="$(var use_sensor_time)"/>
            <param name="configuration_vehicle_length" value="$(var configuration_vehicle_length)"/>
            <param name="configuration_vehicle_width" value="$(var configuration_vehicle_width)"/>
            <param name="configuration_vehicle_height" value="$(var configuration_vehicle_height)"/>
            <param name="configuration_vehicle_wheelbase" value="$(var configuration_vehicle_wheelbase)"/>

            <param name="use_sim_time" value="false"/>
        </node>
    </group>

</launch>
